src/cartesian/CartesianGrid.tsx

```diff
@@ -37,8 +37,14 @@ interface CartesianGridProps extends InternalCartesianGridProps {
   verticalPoints?: number[];
   verticalFill?: string[];
   horizontalFill?: string[];
+  /**
+   * If true, only the lines that correspond to the axes ticks values will be drawn.
+   * If false, extra lines could be added for each axis (at min and max coordinates), if there will not such ticks
+   */
   syncWithTicks?: boolean;
+  /** Array of values, where horizontal lines will be drawn. Numbers or strings, in dependence on axis type */
   horizontalValues?: number[] | string[];
+  /** Array of values, where vertical lines will be drawn. Numbers or strings, in dependence on axis type */
   verticalValues?: number[] | string[];
```

Comment:
How does this interact with 

```
  verticalPoints: {
    description: 'The x-coordinates of all vertical lines.',
    table: {
      type: {
        summary: 'Array',
        defaultValue: [],
      },
      category: 'General',
    },
  },
```

Reply:
What if both are passed? Which one takes precedence?

Reply:
Not required for this PR, but it would be great if you could add comments to the other props as well, and ideally clarify the question above, i.e. declare precedence in the comment, because that would directly land in the documentation. 

Not at all required, I am just thinking out loud.

Reply:
Yeah I'm actually confused about this. Are not `verticalPoints`/`horizontalPoints` the way that the grid internally generates its lines?

Reply:
The priority is: if we have vertical/horizontal points, we use them; then vertical/horizontal values follow; and finally, syncWithTicks, if none of points/values passed.

I added comments about this in CartesianGrid.tsx

Reply:
So, should the type possibly be: 

```
type Config = {
 verticalPoints ..
 horizontalPoints ..
} | {
 verticalValues ..
 horizontalValues ..
} | {
 syncWithTicks ..
}
```

Would that make sense? It could nicely communicate our TS users that using them together does not make sense.

Reply:
@nikolasrieble I think you are right, but I don't know how it could be implemented. I think, my knowledge is not enough.

Reply:
I believe it would be too much to ask from you to fix our typing, especially given that the current typing at Recharts is okay at best. 

I still wanted to share this as the direction I would have preferred.